src/runtime/metal.cpp

```diff
@@ -195,6 +210,11 @@ struct module_state {
 };
 WEAK module_state *state_list = NULL;
 
+// API Capabilities.  If more capabilities need to be checked,
+// this can be refactored to something more robust/general.
+bool halide_metal_api_supports_set_bytes;
```

Comment:
No, you're right-- they should be WEAK, and they should not be exported.  I'll change this & push shortly.